### ISSUE TITLE
Intellisense for C++: Values for enum variables/arguments prediction

### ISSUE THREAD

**Issue by Voha888** [microsoft/vscode-cpptools#5798]
**Type: LanguageService**

**Describe the bug**
- OS and Version: Windows 10 x64
- VS Code Version: 1.47.2
- C/C++ Extension Version: 0.29.0
- Other extensions you installed (and if the issue persists after disabling them): platformio

I'd like to propose improvement to how does IntelliSense operate on enums.

Currently if you have code like following and you press Ctrl+Space inside parentheses of the foo function call, IntelliSense returns huge list of unrelated items like: pre-processor derictives, integer constants, functions. See screenshot attached.

All 99% of the suggested items are not compatible with enum. They cannot be implicitly converted to enum.

**Steps to reproduce**

//

```
// Code example

//
enum SomeEnums
{
Element1,
Element2
};

int main()
{

SomeEnums SomeEnum;

SomeEnum = /* press Ctrl+Space here */

return 0;
}
```

**Screenshots**

The screenshots show the work of another IDE in this situation - the first to offer compatible options
![photo5228978801931824965](https://user-images.githubusercontent.com/17532825/87843417-2f876d00-c8bd-11ea-9289-ed3765717583.jpg)
![photo5228978801931824963](https://user-images.githubusercontent.com/17532825/87843418-30b89a00-c8bd-11ea-8e76-4550ef8bfd1e.jpg)

**Additional context**

**Comment by sean-mcmanus** [comment]
I've filed a feature request on VS. If they add it, we might be able to get the fix from them.

https://developercommunity.visualstudio.com/idea/1122535/intellisense-for-c-values-for-enum-variablesargume-1.html

**Comment by AlmiAlmo** [comment]
I'm not familiar with intellisense and may bullshit. But...

In Visual Studio (and other VS based IDEs like atmel studio) it has been worked for decades. 
![image](https://user-images.githubusercontent.com/97438794/151983160-80f7e0a1-c50d-4c92-a9e3-d2d4d41343b1.png)

**Comment by Alopex95** [comment]
I would appreciate better enum suggestions as well.

**Comment by heartacker** [comment]
This feature has now been released in the latest product update released version 17.3. Please install and let us know your feedback. Please continue to provide feedback and help us be even better.

**Comment by sean-mcmanus** [comment]
@heartacker Yeah, the fix came in too late for our pending 1.12.3 release tomorrow, but it should be available in the following 1.13.0 release afterwards (maybe in a week or so).

**Comment by sean-mcmanus** [comment]
The fix got into VS 17.3, but unfortunately there is more work on the VS Code side to start getting the updated behavior.

**Comment by robinhellmers** [comment]
How is this going, is it still on the map? Seems like it have been fixed for VS according t your link @sean-mcmanus.

> I've filed a feature request on VS. If they add it, we might be able to get the fix from them.
> 
> https://developercommunity.visualstudio.com/idea/1122535/intellisense-for-c-values-for-enum-variablesargume-1.html
